### ISSUE TITLE
Import namespace exports should be immutable

### ISSUE THREAD

**Issue by JsonFreeman** [Microsoft/TypeScript#2456]
We should not allow mutating a property of a namespace import:
```ts
// Library
export var x = 1;

// Consumer
import * as stuff from 'Library';
stuff.x = 0;
```

In ES6, these bindings are immutable.

**Comment by JsonFreeman** [comment]
Why was the fix for this reverted?

**Comment by mhegazy** [comment]
it breaks existing module import syntax. The fix did not handle all cases correctly. we probably need to do it "right" by adding a readonly flag on the type and checking it everywhere.

**Comment by JsonFreeman** [comment]
To clarify, this breaks the `import blah = require(...)` scenario. We'd have to disallow this only for properties of namespaces.

**Comment by Arnavion** [comment]
Fixed by #6532 ( https://github.com/Microsoft/TypeScript/commit/e506378 )?

**Comment by JsonFreeman** [comment]
If the exports on the namespace objects are marked readonly, then yes.
